Inpsyde/Helpers/Names.php

```diff
@@ -0,0 +1,171 @@
+
+final class Names
+{
+    public const NAMEABLE_TOKENS = [
+        T_CLASS,
+        T_NAMESPACE,
+        T_TRAIT,
+        T_INTERFACE,
+        T_CONST,
+        T_FUNCTION,
+        T_VARIABLE,
+        T_ENUM,
+        T_ENUM_CASE,
+    ];
+
+    /**
+     * @param File $file
+     * @param int $position
+     * @return string|null Null is an error, empty string is fine.
+     *
+     * phpcs:disable Generic.Metrics.CyclomaticComplexity
+     */
+    public static function nameableTokenName(File $file, int $position): ?string
+    {
+        // phpcs:enable Generic.Metrics.CyclomaticComplexity
+
+        /** @var array<int, array<string, mixed>> $tokens */
+        $tokens = $file->getTokens();
+        $code = $tokens[$position]['code'] ?? null;
+
+        if (!in_array($code, self::NAMEABLE_TOKENS, true)) {
+            return null;
+        } elseif ($code === T_VARIABLE) {
+            $name = ltrim((string)($tokens[$position]['content'] ?? ''), '$');
+
+            return ($name === '') ? null : $name;
+        }
+
+        if ($code === T_NAMESPACE) {
+            return Namespaces::isDeclaration($file, $position)
+                ? (Namespaces::getDeclaredName($file, $position) ?: '')
+                : null;
+        }
+
+        $namePosition = $file->findNext(T_STRING, $position, null, false, null, true);
+        $name = ($namePosition === false) ? null : (string)$tokens[$namePosition]['content'];
+
+        return ($name === '') ? null : $name;
+    }
+
+    /**
+     * @param File $file
+     * @param int $position
+     * @return string
+     *
+     * phpcs:disable Inpsyde.CodeQuality.FunctionLength
+     * phpcs:disable Generic.Metrics.CyclomaticComplexity
+     */
+    public static function tokenTypeName(File $file, int $position): string
+    {
+        // phpcs:enable Inpsyde.CodeQuality.FunctionLength
+        // phpcs:enable Generic.Metrics.CyclomaticComplexity
+
+        /** @var array<int, array<string, mixed>> $tokens */
+        $tokens = $file->getTokens();
+        $code = $tokens[$position]['code'] ?? -1;
+
+        switch ($code) {
+            case T_CLASS:
+            case T_ANON_CLASS:
+                return 'Class';
+            case T_ENUM:
+                return 'Enum';
+            case T_ENUM_CASE:
+                return 'Enum case';
+            case T_TRAIT:
+                return 'Trait';
+            case T_INTERFACE:
+                return 'Interface';
+            case T_CONST:
+                return 'Constant';
+            case T_FUNCTION:
+                return 'Function';
+            case T_VARIABLE:
+                return Scopes::isOOProperty($file, $position) ? 'Property' : 'Variable';
+            case T_LNUMBER:
+            case T_DNUMBER:
+                return 'Number';
+            case T_STRING:
+                return 'String';
+            case T_THIS:
+                return 'Property';
+            case T_WHITESPACE:
+                return 'White space';
+        }
```

Comment:
I don't think performance would really increase in any noticeable way, if any at all. And because the case of "variable" should be treated separately, then also the overall readability won't be much better.
Not to mention that if another case similar to the variable would emerge, then it also would need to be treated separately.
I would refactor to `match` if this would be PHP 8+.